pkg/controllers/vdb/onlineupgrade_reconciler.go

```diff
@@ -941,20 +1010,50 @@ func (r *OnlineUpgradeReconciler) redirectConnectionsToReplicaGroupB(ctx context
 		return ctrl.Result{}, err
 	}
 
-	// In lieu of the redirect, we are simply going to update the service object
-	// to map to nodes in replica group B. There is no state to check to avoid
-	// this function. The updates themselves are idempotent and will simply be
-	// no-op if already done.
-	// Routing is easy since the sandbox is a copy of the main cluster.
-	//
+	// remove client routing labels from replica group a
+	actor := MakeClientRoutingLabelReconciler(r.VRec, r.Log, r.VDB, r.PFacts[vapi.MainCluster], DrainNodeApplyMethod, "")
+	r.Manager.traceActorReconcile(actor)
+	res, err := actor.Reconcile(ctx, &ctrl.Request{})
+	if verrors.IsReconcileAborted(res, err) {
+		return res, err
+	}
 	// Add the client routing labels to pods in the target subcluster. This
 	// ensures the service object can reach them.  We use the podfacts for the
 	// sandbox as we will always route to pods in the sandbox.
-	actor := MakeClientRoutingLabelReconciler(r.VRec, r.Log, r.VDB, sbPFacts,
-		AddNodeApplyMethod, "")
+	actor = MakeClientRoutingLabelReconciler(r.VRec, r.Log, r.VDB, sbPFacts, AddNodeApplyMethod, "")
 	r.Manager.traceActorReconcile(actor)
+	if res, err = actor.Reconcile(ctx, &ctrl.Request{}); verrors.IsReconcileAborted(res, err) {
+		return res, err
+	}
 
-	return actor.Reconcile(ctx, &ctrl.Request{})
+	initiator, ok := r.PFacts[vapi.MainCluster].findFirstUpPod(false /*not allow read-only*/, "" /*arbitrary subcluster*/)
+	if !ok {
+		r.Log.Info("No Up nodes found; requeueing reconciliation")
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	scMap := r.VDB.GenSubclusterMap()
+	// after modifying routing to the sandbox, redirect existing connections to the sandbox too
+	for i := range r.VDB.Spec.Subclusters {
```

Comment:
What connection are you riderecting during each iteration?

Reply:
each iteration tells the given subcluster to redirect connections